File: Pangrams.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'pangrams' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */


string pangrams(string s)
{
    std::vector<bool> alphabet(26, false);
    int count = 0;

    for (char c : s)
    {
        if (std::isalpha(c))
        {
            int index = std::tolower(c) - 'a';
            if (!alphabet[index])
            {
                alphabet[index] = true;
                if (++count == 26) return "pangram";
            }
        }
    }

    return "not pangram";
}


string pangrams2(string s)
{
    std::unordered_set<char> st;
    
    for (char c : s)
    {
        if (std::isalpha(c))
        {
            st.insert(std::tolower(c));
            if (st.size() == 26) return "pangram";
        }
    }

    return "not pangram";
}


string pangrams3(string s)
{
    std::unordered_map<char, bool> mp;
    int count = 0;

    for (char c : s)
    {
        if (std::isalpha(c))
        {
            c = std::tolower(c);
            if (!mp[c])
            {
                mp[c] = true;
                if (++count == 26) return "pangram";
            }
        }
    }

    return "not pangram";
}


int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = pangrams(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Pangrams.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'pangrams' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static String pangrams(String s) {
        boolean[] alphabet = new boolean[26];
        int count = 0;

        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) {
                int index = Character.toLowerCase(c) - 'a';
                if (!alphabet[index]) {
                    alphabet[index] = true;
                    if (++count == 26) return "pangram";
                }
            }
        }

        return "not pangram";
    }
    
    
    public static String pangrams2(String s) {
        Set<Character> set = new HashSet<>();

        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) {
                set.add(Character.toLowerCase(c));
                if (set.size() == 26) return "pangram";
            }
        }

        return "not pangram";
    }
    
    
    public static String pangrams3(String s) {
        Map<Character, Boolean> map = new HashMap<>();
        int count = 0;

        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) {
                char lowerC = Character.toLowerCase(c);
                if (!map.containsKey(lowerC)) {
                    map.put(lowerC, true);
                    if (++count == 26) return "pangram";
                }
            }
        }

        return "not pangram";
    }    
}


public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = bufferedReader.readLine();

        String result = Result.pangrams(s);

        bufferedWriter.write(result);
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
